cruft/_commands/utils/generate.py

```diff
@@ -0,0 +1,120 @@
+import json
+import os
+from pathlib import Path
+from shutil import move, rmtree
+from tempfile import TemporaryDirectory
+from typing import Any, Dict, Optional, Set
+
+from cookiecutter.generate import generate_files
+from git import Repo
+
+from .cookiecutter import CookiecutterContext, generate_cookiecutter_context
+from .cruft import get_cruft_file
+
+try:
+    import toml  # type: ignore
+except ImportError:  # pragma: no cover
+    toml = None  # type: ignore
+
+CruftState = Dict[str, Any]
+
+
+def cookiecutter_template(
+    output_dir: Path,
+    repo: Repo,
+    project_dir: Path = Path("."),
+    cookiecutter_input: bool = False,
+    checkout: Optional[str] = None,
+) -> CookiecutterContext:
+    """Generate a clean cookiecutter template in output_dir."""
+    pyproject_file = project_dir / "pyproject.toml"
+    cruft_file = get_cruft_file(project_dir)
```

Comment:
Would it be possible to add a cruft state argument to the function and pass it directly? Otherwise this currently leads to the cruft state being read twice - once in the relevant CLI commands and once here.

Reply:
Yup, nice catch actually :monocle_face: . Fixed it.